ARM-wvd-templates/DSC/Configuration.ps1

```diff
@@ -106,5 +109,15 @@ configuration AddSessionHost
                 }
             }
         }
+        
+        if ($true -eq $AadJoin) {
+            Registry RDAgentAADJPrivateConfig {
+                DependsOn = "[Script]ExecuteRdAgentInstall$(if ($rdshIsServer) { 'Server' } else { 'Client' })"

```

Comment:
What does this line do? where is the script "ExecuteRdAgentInstall"? This variable "$rdshIsServer" is assigned by isRdshServer, where is this method "isRdshServer"?

Reply:
@nakranimohit0 any insight?

Reply:
@shanberg1 @philipktlin Documentation for DSC Registry block: https://docs.microsoft.com/en-us/powershell/scripting/dsc/reference/resources/windows/registryresource?view=powershell-5.1#common-properties

DependsOn basically defines the order at which configurations are executed. In this case it depends on DSC script `ExecuteRdAgentInstallServer` / `ExecuteRdAgentInstallClient` block which are defined above. `$rdshIsServer` is a variable that is also define above using method `isRdshServer`. Method `isRdshServer` is defined in `Functions.ps1`